libcxx/include/__overridable_function

```diff
@@ -0,0 +1,38 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCPP___OVERRIDABLE_FUNCTION
+#define _LIBCPP___OVERRIDABLE_FUNCTION
+
+#include <__config>
+#include <cstdint>
+
+_LIBCPP_BEGIN_NAMESPACE_STD
+
+#define _LIBCPP_MAKE_OVERRIDABLE_FUNCTION_DETECTABLE                                                                   \
+  __attribute__((__section__("__TEXT,__lcxx_override,regular,pure_instructions")))
```

Comment:
Yeah, that's right. In a way, this is "fortunate" since this whole `__overridable_function` thing is only used to power the assertion we have in case a user forgot to override the `nothrow` version. So we could just disable it on some platforms -- but in practice it would be really nice to have this diagnostic on all supported platforms, since this is so tricky.

Reply:
To port this to ELF, I think you could use `__attribute__((section("lcxx_override")))`, and then declare external `__start_lcxx_override` and `__stop_lcxx_override` symbols. That should work with all the linkers we care about. COFF can probably be done, but I'm sure it's more involved and probably not worth doing.

Reply:
I think I finally got this to work for ELF! The next revision will tell.